contracts/metatx/ERC2771Context.sol

```diff
@@ -18,27 +18,54 @@ abstract contract ERC2771Context is Context {
     /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
     address private immutable _trustedForwarder;
 
+    /**
+     * @dev Sets an immutable trusted forwarder.
+     *
+     * NOTE: The trusted forwarder can be replaced by overriding {trustedForwarder}.
+     */
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(address trustedForwarder) {
-        _trustedForwarder = trustedForwarder;
+    constructor(address trustedForwarder_) {
+        _trustedForwarder = trustedForwarder_;
     }
 
+    /**
+     * @dev Returns the address of the trusted forwarder.
+     */
+    function trustedForwarder() public view virtual returns (address) {
+        return _trustedForwarder;
+    }
+
+    /**
+     * @dev Indicates whether any particular address is the trusted forwarder.
+     */
     function isTrustedForwarder(address forwarder) public view virtual returns (bool) {
-        return forwarder == _trustedForwarder;
+        return forwarder == trustedForwarder();
     }
 
-    function _msgSender() internal view virtual override returns (address sender) {
+    /**
+     * @dev Override for `msg.data`. Defaults to the original `msg.sender` whenever
+     * a call is not performed by the trusted forwarder or the calldata length is less than
+     * 20 bytes (an address length).
+     */
+    function _msgSender() internal view virtual override returns (address) {
         if (isTrustedForwarder(msg.sender) && msg.data.length >= 20) {
+            address sender;
```

Comment:
What is the reason to remove the named return value? It makes the code nicer. We tend to use this pattern when the return value is set in assembly.

Reply:
No concrete reason is provided for the issue other than what the title describes ("Unused Named Return Variables"). I'd agree with reverting this change.

Reply:
Yeah let's revert. This variable is used.

Reply:
Sure thing, removed in [58546a1](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4502/commits/58546a17c4e78aedf55aa3ae49e81d421bdca4a6)